### PR TITLE
Proposal: change web client build process

### PR DESCRIPTION
**Pull request by zachmullen**

### CODE REVIEW

**Comment by zachmullen** [comment]
I've created the 2.0-integration branch and made this PR against it, so this is now ready for review and can be merged.

**Comment by zachmullen** [comment]
@opadron PTAL

File: docs/installation.rst [hunk 1]
```diff
@@ -115,7 +115,7 @@ to install the plugins as well.
 To build the client-side code project, cd into the root of the repository
 and run: ::
 
-    npm install
+    npm install && npm run build
 
 This will run multiple `Grunt <http://gruntjs.com>`_ tasks, to build all of
```
Comment (opadron) [inline docs/installation.rst]: We're still using Grunt, right?  Either way, we should make sure to keep a close look at the docs as we make bigger changes.

File: grunt_tasks/package.js [hunk 2]
```diff
@@ -38,8 +38,8 @@ module.exports = function (grunt) {
 
                         var fname = 'girder-' + grunt.config.get('pkg').version +
                             '.tar.gz';
-                        var stat = fs.statSync(fname);
-                        if (!err && stat.isFile() && stat.size > 0) {
+                        var stat = fs.existsSync(fname) && fs.statSync(fname);
+                        if (!err && stat && stat.isFile() && stat.size > 0) {
```
Comment (opadron) [inline grunt_tasks/package.js]: I'm not too familiar with this part.  Are these changes related to the build process?

Comment (zachmullen) [inline grunt_tasks/package.js]: This is kind of unrelated, it just fixes a bug in the case that packaging fails. Trying to `stat` a nonexistent file throws an exception.

File: grunt_tasks/plugin.js [hunk 3]
```diff
@@ -246,25 +211,6 @@ module.exports = function (grunt) {
...
-            this.requiresConfig(tasks);
-
-            // queue the install tasks
-            grunt.config.get(tasks).forEach(function (task) {
-                grunt.task.run(task);
-            });
-        }
-    );
-
-    /**
```
Comment (opadron) [inline grunt_tasks/plugin.js]: Do these removals mean that plugins won't be able to use their own `packages.json` file, anymore?

Comment (zachmullen) [inline grunt_tasks/plugin.js]: No, this is now handled in the postinstall step by a new script.

File: docs/installation.rst (hunk 1 above)
Comment (zachmullen) [inline docs/installation.rst]: Yes, this branch is just changing the build commands, not the underlying technologies.

**Comment by opadron** [comment]
LGTM
